driver/src/main/java/org/neo4j/driver/internal/security/TLSSocketChannel.java

```diff
@@ -345,18 +345,22 @@ private HandshakeStatus wrap( ByteBuffer buffer ) throws IOException, ClientExce
             }
             else
             {
+                logger.debug( "Network output buffer doesn't need enlarging, flushing data to the channel instead to open up space on the buffer." );
                 // flush as much data as possible
                 cipherOut.flip();
-                if ( channelWrite( cipherOut ) == 0 )
-                {
-                    throw new ClientException( format(
-                                    "Failed to enlarge network buffer from %s to %s. This is either because the " +
-                                    "new size is however less than the old size, or because the application " +
-                                    "buffer size %s is so big that the application data still cannot fit into the " +
-                                    "new network buffer.", curNetSize, netSize, buffer.capacity() ) );
+                while ( cipherOut.hasRemaining() ) {
```

Comment:
Makes sense, and if socket is completely broken exception will be thrown by `#channelWrite ()` method.